### PR TITLE
XML attribute associated to wrong type from XSD

### PR DESCRIPTION
**Pull request by angelozerr**
XML attribute associated to wrong type from XSD

Fixes https://github.com/redhat-developer/vscode-xml/issues/524

### CODE REVIEW

**Comment by angelozerr** [comment]
@datho7561 @JessicaJHee if you have time please review this PR. It is easy to test it.

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaDiagnosticsTest.java [hunk 1]
```diff
@@ -509,6 +509,18 @@ public void cvc_datatype_valid_1_2_3Empty() throws Exception {
 				d(1, 1, 1, 10, XMLSchemaErrorCode.cvc_type_3_1_3));
 	}
 
+	@Test
```
Comment (JessicaJHee) [inline org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaDiagnosticsTest.java:512]: Would be helpful to link the original issue for this test case

**Review by JessicaJHee — COMMENTED** [review]
Solves the issue as expected! Just had one small suggestion

File: org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaDiagnosticsTest.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaDiagnosticsTest.java:512]: fixed

**Comment by angelozerr** [comment]
Thanks @JessicaJHee for your review.

**Pull request merged**